src/com/phonebookservice/database/ContactsDatabase.java

```diff
@@ -0,0 +1,69 @@
+package com.phonebookservice.database;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.phonebookservice.model.Contact;
+import com.phonebookservice.util.MyArrayList;
+
+public final class ContactsDatabase {
+    private static ContactsDatabase singleton;
+    private final ContactsFileParser parser;
+    private final Collection<Contact> contactList;
+    private final Map<String, Contact> idToContactMap;
+
+    /**
+     * Contacts database constructor.
+     *
+     * @param fileName the file name.
+     */
+    private ContactsDatabase(final String fileName) {
+        this.contactList = new MyArrayList<Contact>();
+        this.idToContactMap = new HashMap<>();
+        this.parser = new ContactsFileParser(fileName);
```

Comment:
Variable name should be
ContactFileParser

Reply:
why ? it read all contacts and write all contacts

Reply:
You named it as a parser. Parser of what ??